src/Network/Network/LoadBalancer/SetAzureLoadBalancerBackendPool.cs

```diff
@@ -0,0 +1,211 @@
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Set, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "LoadBalancerBackendAddressPool"), OutputType(typeof(PSBackendAddressPool))]
+    public partial class SetAzureLoadBalancerBackendPool : NetworkBaseCmdlet
+    {
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
+        private const string SetByParentObjectParameterSet = "SetByParentObjectParameterSet";
+        private const string SetByInputObjectParameterSet = "SetByInputObjectParameterSet";
+        private const string SetByResourceIdParameterSet = "SetByResourceIdParameterSet";
+
+        [Parameter(Mandatory = true, HelpMessage = "The resource group name of the load balancer.", ParameterSetName = SetByNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+
+        [Parameter(Mandatory = true,HelpMessage = "The name of the load balancer.",  ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string LoadBalancerName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the backend pool.", ParameterSetName = SetByNameParameterSet)]
+        [Parameter(Mandatory = true, HelpMessage = "The name of the backend pool.", ParameterSetName = SetByParentObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The load balancer resource.", ValueFromPipeline = true, ParameterSetName = SetByParentObjectParameterSet)]
+        public PSLoadBalancer LoadBalancer { get; set; }
+
+
+        [Parameter(Mandatory = true, HelpMessage = "The backend address pool to set", ParameterSetName = SetByInputObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSBackendAddressPool InputObject { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The backend addresses.", ParameterSetName = SetByNameParameterSet)]
+        [Parameter(Mandatory = true, HelpMessage = "The backend addresses.", ParameterSetName = SetByParentObjectParameterSet)]
+        [Parameter(Mandatory = true, HelpMessage = "The backend addresses.", ParameterSetName = SetByResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSLoadBalancerBackendAddress[] LoadBalancerBackendAddress { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = SetByResourceIdParameterSet)]
```

Comment:
should we also support parent resource ID?

Reply:
I don't see a need for this.